core/src/main/java/co/cask/wrangler/steps/transformation/TableLookup.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.steps.transformation;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.DatasetInstantiationException;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.etl.api.Lookup;
+import co.cask.wrangler.api.AbstractStep;
+import co.cask.wrangler.api.PipelineContext;
+import co.cask.wrangler.api.Record;
+import co.cask.wrangler.api.StepException;
+import co.cask.wrangler.api.Usage;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An AbstractStep that performs a lookup into a Table Dataset and adds the row values into the record.
+ */
+@Usage(
+  directive = "table-lookup",
+  usage = "table-lookup <column> <table>",
+  description = "Uses the given column as a key to perform a lookup into the specified table."
+)
+public class TableLookup extends AbstractStep {
+
+  private final String column;
+  private final String table;
+
+  private boolean initialized;
+  private co.cask.cdap.etl.api.lookup.TableLookup tableLookup;
```

Comment:
This is referencing the `TableLookup` interface in CDAP ETL-API, instead of the class in which it is located, the `TableLookup` directive.